### DISCUSSION TITLE
Aliases (chain = flatMap)

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [gcanti/fp-ts#1844] [Ideas]
Hi.

I just learned about this library, and I got interested, but when I was watching a tutorial on YouTube, the author explained the chain function as a replacement for map, that flattens the options, and the author wrote "flatMap" to explain it better.

So I wondered "why is it named chain instead of flatMap?". I prefer consistency and this is bit of annoying.
And then I wondered what other stuff is being named in a non consistent way.
I'm just trying to understand the logic behind the naming decisions, not bringing negativity. I am really hyped for this library.

So my suggestion is, if due to backward compatibility there's no chance of renaming it, then is it possible to just tree-shake-import the aliases, so we can use flatMap instead of chain? Many libraries have aliases.
pipe is already associated with chain, chainable functions.
flatMap sounds more like what Array.flat does.

Any kind response will be appreciated.

**Comment by gcanti** [comment]
> why is it named chain instead of flatMap?

The choice dates back a few years ago when the only specification regarding functional programming in JavaScript was [fantasy land].(https://github.com/fantasyland/fantasy-land)

Regarding `flatMap` it's definitely one of the things I want to backport from [`@effect/data`](https://github.com/Effect-TS/data).

**Reply by angelhdzmultimedia** [reply]
> > why is it named chain instead of flatMap?
> 
> The choice dates back a few years ago when the only specification regarding functional programming in JavaScript was [fantasy land].(https://github.com/fantasyland/fantasy-land)
> 
> Regarding `flatMap` it's definitely one of the things I want to backport from [`@effect/data`](https://github.com/Effect-TS/data).

Thank you for your time and information provided. Much clear now.

**Reply by gcanti** [reply]
https://github.com/gcanti/fp-ts/pull/1849

**Reply by angelhdzmultimedia** [reply]
Wow that was fast!  I'm gonna have fun learning this lib tonight!

Wishing you health, peace, and success.
